### PR TITLE
Fix some broken links, add Go applications to index.md

### PR DESCRIPTION
**Pull request by hkjn**
As an aside, we might want to set up some quick scraper as part of CI to check that there's no links that 404s, since checking manually might become cumbersome..

### CODE REVIEW

File: docs/os/build-details.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ It performs the following steps:
 1. cloning the [`github.com/armbian/build`](https://github.com/armbian/build/) repo into `armbian/armbian-build/`, if it doesn't exist already
 1. copying over the following files from the host system into `armbian/armbian-build/userpatches/overlay/`:
     1. `armbian/base/`: scripts and configs included in the Armbian image
-    1. `build/`: contains all [built Go binaries](/docs/go-apps.md) and their associated `.service` files
+    1. `build/`: contains all [built Go binaries](go-apps.md) and their associated `.service` files
```
Comment (Stadicus) [inline docs/os/build-details.md]: Not sure that works out of the `/docs/os/` directory?

Comment (hkjn) [reply docs/os/build-details.md]: Yeah, I think you're right. On the live site it seems that the URL is https://base.shiftcrypto.ch/go, so I've updated this accordingly. It seems to render as `http://localhost:4000/go` on local Jekyll, so I think the new updated link should work.

**Comment by hkjn** [comment]
Thanks for the review @Stadicus, PTAL.

**Comment by Stadicus** [comment]
utACK https://github.com/digitalbitbox/bitbox-base/commit/03f090a6bf1ea1cc0c64f0c59c7d6c1bcfdd80cf

**Pull request merged**